### PR TITLE
Add apply method

### PR DESCRIPTION
**Pull request by hanny24**

### CODE REVIEW

**Comment by hanny24** [comment]
@tomasherman @jakubjanecek

File: src/main/scala/com/avast/utils2/Done.scala [hunk 1]
```diff
@@ -7,4 +7,6 @@ package com.avast.utils2
   */
 sealed trait Done
 
-case object Done extends Done
+case object Done extends Done {
+  def apply(): Done = this
```
Comment (jakubjanecek) [inline src/main/scala/com/avast/utils2/Done.scala:11]: Why?

Comment (tomasherman) [reply src/main/scala/com/avast/utils2/Done.scala:11]: yes, why?!

Comment (hanny24) [reply src/main/scala/com/avast/utils2/Done.scala:11]: Because it sucks to type `Done: Done`

Comment (jakubjanecek) [reply src/main/scala/com/avast/utils2/Done.scala:11]: But doesn't this mean you have to type `Done()`? But I guess it can be added.

**Review by jakubjanecek — APPROVED** [review]

**Comment by hanny24** [comment]
It's a shame I have no permissions to merge this.

**Pull request merged**